includes/class-convertkit-html-parser.php

```diff
@@ -67,9 +67,14 @@ public function __construct( $content, $flags = false ) {
 	 */
 	public function get_body_html() {
 
-		// Return modified content in the <body> tag.
-		preg_match( '/<body[^>]*>(.*?)<\/body>/is', $this->html->saveHTML(), $matches );
-		return $matches[1] ?? '';
+		$body = $this->html->getElementsByTagName( 'body' )->item( 0 );
+
+		$html = '';
+		foreach ( $body->childNodes as $child ) { // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+			$html .= $this->html->saveHTML( $child );
```

Comment:
@noelherrick You pretty much had this sorted, but I didn't [spot the difference with using `saveXML`](https://github.com/Kit/convertkit-wordpress/pull/879#discussion_r2285900462). Using `saveHTML` does the trick. Encoding works as it's already set in the `html` DOMDocument instance loaded in the construct.

Reply:
Thank you so much for doing this! I know it's not the biggest deal, but I wanted to reduce fragility and what I've seen is that code and patterns get copied throughout a codebase so I'd like to have a good example!